src/Akeneo/Pim/Enrichment/Product/back/API/Command/UserIntent/SetDateValue.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Product\API\Command\UserIntent;
+
+/**
+ * @copyright 2022 Akeneo SAS (https://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class SetDateValue implements ValueUserIntent
+{
+    public function __construct(
+        private string $attributeCode,
+        private ?string $channelCode,
+        private ?string $localeCode,
+        private \DateTime $value
```

Comment:
```suggestion
        private \DateTimeInterface $value
```

Using interface is better